software-development/SDLC.md

```diff
@@ -0,0 +1,101 @@
+    o	In a complex system there are multiple applications communicating with each other to complete a business process<br/>
+    o	What data format will be used<br/>
+
+### Software architecture
+
+Because of its importance we spend a little bit more time on the software architecture as it is an essential part of software development. 
+As described on [this website on architecture](https://martinfowler.com/architecture/), Architecture is described as the way the highest level components are wired together. The site also explains the reason why architecture matters:
+
+*Architecture is a tricky subject for the customers and users of software products - as it isn't something they immediately perceive. But a poor architecture is a major contributor to the growth of cruft - elements of the software that impede the ability of developers to understand the software. Software that contains a lot of cruft is much harder to modify, leading to features that arrive more slowly and with more defects.*
+
+It's important to understand that making fundamental changes to the software architecture can be costly; changing from a MongoDB database to Postgress can be complex and time consuming. Therefore it's important to understand that: <br/>
+*Software architecture is about making fundamental structural choices that are costly to change once implemented.*
+
+
+
+**Output of this phase:** a software design <br/>
+**Persons involved:** Engineers, either a (senior) software engineer of the team or an architect<br/>
+
+## Software Development
+**Background**: In this phase the actual coding starts and the system is build. One or multiple engineer start to build the entire system writing code in a chosen programming language. The requirements and the high level architecture design usually serve as input to this phase. As there are usually more than one engineer involved, and in the future other engineers might need to work on the code you created it’s common that the engineers follow the coding guidelines of the company they are working for. 
+During development sometimes short cuts are taken to get faster to market. These shortcut are sometimes called technical debt. Wikipedia states; Technical Debt is a concept in software development that reflects the implied cost of additional rework caused by choosing an easy (limited) solution now instead of using a better approach that would take longer.<br/>
+
+**Who is responsible**: Engineers
+
+
+## Testing
+**Background:** Testing is mentioned as a separate phase but usually already starts earlier. Some tests are written during development to automate some of the testing. In general in this phase the requirements used as the starting point to test the functionality of the software. Any defects found are assigned to developers to get them fixed. As described above the testing is an integral part of development phase more and more as tests can be automated and written in code.
```

Comment:
`Tests are written during development to automate some of the testing` instead of `Some tests are written during development to automate some of the testing`
`Generally in this phase the requirements used as the starting point of the application are used to test the functionality of the software` instead of `In general in this phase the requirements used as the starting point to test the functionality of the software.`
`As described above, testing is an integral part of the development phase as more and more tests can be automated` instead of `As described above the testing is an integral part of development phase more and more as test can be automated`